### ISSUE TITLE
Instance field not found

### ISSUE THREAD

**Issue by amahdy** [vaadin/framework#8986]
Vaadin version: 8.0.5

```java
public class Person {
    LocalDate mydate;
    public LocalDate getMydate() { ... }
    public void setMydate(...) { … }
}
```

```java
private TextField mydate = new TextField();

init(...) {
  Binder<Person> binder = new Binder<>(Person.class);
  binder.forMemberField(mydate)
    .withConverter(str -> LocalDate.now(), date -> "Hello");
  binder.bindInstanceFields(this);
}
```

Getting error:
`java.lang.IllegalStateException: There are no instance fields found for automatic binding`

**Comment by homer54** [comment]
The exception "There are no instance fields found for automatic binding" shouldn't be raised.
It's hard to figure out if the bindInstanceFields must be called or if the code has already bind all fields.

**Comment by ToArnld** [comment]
Same problem with 8.1.0.alpha4

When all instance fields are already bound, `bindInstanceFields` throws this error.

**Comment by mstahv** [comment]
Here is a full code snippet that can be used to reproduce the issue:

    public class MyUI extends UI {
    
        private TextField mydate = new TextField();
        private TextField string = new TextField();
    
        @Override
        protected void init(VaadinRequest request) {
            Binder<Person> binder = new Binder<>(Person.class);
            binder.forMemberField(mydate)
                    .withConverter(str -> LocalDate.now(), date -> "Hello");
            binder.bindInstanceFields(this);
    
            Person p = new Person();
            binder.setBean(p);
    
            setContent(new VerticalLayout(
                    mydate,
                    string
            ));
    
            binder.addValueChangeListener(e-> Notification.show(p.toString()));
        }
    
        public static class Person {
            private LocalDate mydate;
            private String string; // TODO Comment this field out and it breaks
    
            public LocalDate getMydate() {
                return mydate;
            }
    
            public void setMydate(LocalDate mydate) {
                this.mydate = mydate;
            }
    
            public String getString() {
                return string;
            }
    
            public void setString(String string) {
                this.string = string;
            }
    
            @Override
            public String toString() {
                return "Person{" +
                        "mydate=" + mydate +
                        '}';
            }
        }
    
    }

This is a nasty bug, please fix! Verified to exist in latest 8.1-SNAPSHOT

**Comment by tsuoanttila** [comment]
This was picked to 8.0.7

**Comment by zygimantus** [comment]
Is it fixed? I am getting the same error in Vaadin 8.1.7 when setting an entity for a class that extends AbstractForm .

**Comment by 452** [comment]
reduced, with spring initializer
<vaadin.version>13.0.3</vaadin.version>
https://spring.io/guides/gs/crud-with-vaadin
